tests/taproot_maker_abort2.rs

```diff
@@ -0,0 +1,210 @@
+
+    // Fund the Taproot Maker with 4 UTXOs of 0.05 BTC each
+    fund_taproot_makers(
+        &taproot_makers,
+        bitcoind,
+        4,
+        Amount::from_btc(0.05).unwrap(),
+    );
+
+    // Start the Taproot Maker Server thread
+    info!("🚀 Initiating Maker server...");
+
+    let taproot_maker_threads = taproot_makers
+        .iter()
+        .map(|maker| {
+            let maker_clone = maker.clone();
+            thread::spawn(move || {
+                start_maker_server_taproot(maker_clone).unwrap();
+            })
+        })
+        .collect::<Vec<_>>();
+
+    // Wait for taproot maker to complete setup
+    for maker in &taproot_makers {
+        while !maker.is_setup_complete.load(Relaxed) {
+            info!("⏳ Waiting for taproot maker setup completion");
+            thread::sleep(Duration::from_secs(10));
+        }
+    }
+
+    // Sync wallets after setup
+    for maker in &taproot_makers {
+        maker.wallet().write().unwrap().sync_and_save().unwrap();
+    }
+    let actual_maker_spendable_balances = verify_maker_pre_swap_balance_taproot(&taproot_makers);
+    info!("🔄 Initiating taproot maker abort 2 case. (Will fail mid-swap due to one maker closing connection at handling private key step)");
+
+    // Swap params - small amount for faster testing
+    let swap_params = SwapParams {
+        send_amount: Amount::from_sat(500000), // 0.005 BTC
+        maker_count: 2,
+        tx_count: 3,
+        required_confirms: 1,
+        manually_selected_outpoints: None,
+    };
+
+    // Attempt the swap - it will fail when maker closes connection
+    // After recovery, do_coinswap returns Ok(None) to indicate recovery was triggered
+    match taproot_taker.do_coinswap(swap_params) {
+        Ok(Some(_report)) => {
+            panic!("Swap should have failed due to maker closing connection, but succeeded with report!");
+        }
+        Ok(None) => {
+            info!("✅ Taproot coinswap triggered recovery as expected (Ok(None))");
+        }
+        Err(e) => {
+            info!("✅ Taproot coinswap failed as expected: {:?}", e);
+        }
+    }
+
+    // Mine blocks to confirm any recovery transactions
+    info!("⛏️ Mining blocks to confirm recovery transactions...");
+    generate_blocks(bitcoind, 2);
+    taproot_taker.get_wallet_mut().sync_and_save().unwrap();
+
+    info!("📊 Taproot Taker balance after hashlock recovery:");
+    let taker_balances_after = taproot_taker.get_wallet().get_balances().unwrap();
+    info!(
+        "  Regular: {}, Contract: {}, Spendable: {}",
+        taker_balances_after.regular, taker_balances_after.contract, taker_balances_after.spendable
+    );
+
+    // Now wait for maker to extract preimage and recover via hashlock
+    info!("⏳ Waiting for maker to extract preimage and recover via hashlock...");
+    std::thread::sleep(std::time::Duration::from_secs(60));
+    // Mine more blocks to give maker time to see the hashlock sweep
+    generate_blocks(bitcoind, 2);
```

Comment:
Instead of sleeping, we should actually watch for the right log. We already have `assert_log()` for the test framework. Made exactly for these behavior assertions. 

Applies to all other places where tests are sleeping for something to get done that can be identified in the logs.